src/weights.jl

```diff
@@ -569,47 +565,62 @@ function quantile(v::RealVector{V}, w::AbstractWeights{W}, p::RealVector) where
         x < 0 && error("weight vector cannot contain negative entries")
     end
 
-    # full sort
-    vw = sort!(collect(zip(v, w.values)))
+    wvalues = w.values
+    nz = find(w.values)
```

Comment:
Use `find(!iszero, w.values)` since the previous form is deprecated on 0.7. Or probably even better, just do `nz = .!iszero.(w.values)`, since a boolean vector will be more efficient (you know the size in advance and it's smaller).

Also move this below to the place where it's used for clarity.

Finally, why is that operation needed for frequency weights? Shouldn't the algorithm be able to skip entries with zero weights on its own?

Reply:
No, the algorithm does not work with zero weights in its current form. There is an issue for instance if the highest value has zero weight. The algorithm also only keeps track of the last visited value, instead of the last visiting value with non zero weight.